src/main/java/leduc/command/DeadlineCommand.java

```diff
@@ -55,14 +57,31 @@ public void execute(TaskList tasks, Ui ui, Storage storage)
             throw new EmptyDeadlineDateException();
         } else {
             String description = taskDescription[0].trim();
-            String deadlineString = taskDescription[1].trim();
+            String[] prioritySplit = taskDescription[1].trim().split("prio");
+            String deadlineString = prioritySplit[0].trim();
             //date format used: dd/MM/yyyy HH:mm
             if (deadlineString.isBlank()) {
                 throw new EmptyDeadlineDateException();
             }
             else {
                 Date d = new Date(deadlineString);
-                DeadlinesTask newTask = new DeadlinesTask(description, d);
+                DeadlinesTask newTask = null;
+                if (prioritySplit.length == 1){
+                    newTask = new DeadlinesTask(description, d);
+                }
+                else {
+                    int priority = -1 ;
+                    try{
+                        priority = Integer.parseInt(prioritySplit[1].trim());
+                    }
+                    catch(Exception e){
```

Comment:
If it cannot be parse to an int, why not use PrioritizeFormatException

Reply:
A specified that PrioritizeLimitException represent a exception when the new priority is not an int or is greater than 9 or less than 0.

And PrioritizeFormatException is more for when the format for the PrioritizeCommand is not respected.